hbase-server/src/main/java/org/apache/hadoop/hbase/security/token/TokenUtil.java

```diff
@@ -42,81 +39,11 @@
  */
 @InterfaceAudience.Public
 public class TokenUtil {
+
   // This class is referenced indirectly by User out in common; instances are created by reflection
   private static final Logger LOG = LoggerFactory.getLogger(TokenUtil.class);
 
-    /**
-     * See {@link ClientTokenUtil#obtainToken(org.apache.hadoop.hbase.client.AsyncConnection)}.
-     * @deprecated External users should not use this method. Please post on
-     *   the HBase dev mailing list if you need this method. Internal
-     *   HBase code should use {@link ClientTokenUtil} instead.
-     */
-  @Deprecated
-  public static CompletableFuture<Token<AuthenticationTokenIdentifier>> obtainToken(
-      AsyncConnection conn) {
-    return ClientTokenUtil.obtainToken(conn);
-  }
-
-  /**
-   * See {@link ClientTokenUtil#obtainToken(org.apache.hadoop.hbase.client.Connection)}.
-   * @deprecated External users should not use this method. Please post on
-   *   the HBase dev mailing list if you need this method. Internal
-   *   HBase code should use {@link ClientTokenUtil} instead.
-   */
-  @Deprecated
-  public static Token<AuthenticationTokenIdentifier> obtainToken(Connection conn)
-      throws IOException {
-    return ClientTokenUtil.obtainToken(conn);
-  }
-
-
-  /**
-   * See {@link ClientTokenUtil#toToken(org.apache.hadoop.security.token.Token)}.
-   * @deprecated External users should not use this method. Please post on
-   *   the HBase dev mailing list if you need this method. Internal
-   *   HBase code should use {@link ClientTokenUtil} instead.
-   */
-  @Deprecated
-  public static AuthenticationProtos.Token toToken(Token<AuthenticationTokenIdentifier> token) {
-    return ClientTokenUtil.toToken(token);
-  }
-
-  /**
-   * See {@link ClientTokenUtil#obtainToken(org.apache.hadoop.hbase.client.Connection,
-   * org.apache.hadoop.hbase.security.User)}.
-   * @deprecated External users should not use this method. Please post on
-   *   the HBase dev mailing list if you need this method. Internal
-   *   HBase code should use {@link ClientTokenUtil} instead.
-   */
-  @Deprecated
-  public static Token<AuthenticationTokenIdentifier> obtainToken(
-      final Connection conn, User user) throws IOException, InterruptedException {
-    return ClientTokenUtil.obtainToken(conn, user);
-  }
-
-  /**
-   * See {@link ClientTokenUtil#obtainAndCacheToken(org.apache.hadoop.hbase.client.Connection,
-   * org.apache.hadoop.hbase.security.User)}.
-   */
-  public static void obtainAndCacheToken(final Connection conn,
```

Comment:
This method is not deprecated. Could you put it back and add a deprecation note with removal in 4.0.0?

Reply:
I was thinking if it would be better for client to use `ClientTokenUtil.obtainAndCacheToken()` instead.
But yes since it is not yet deprecated and it is IA.public, better to put back

Reply:
@petersomogyi updated the PR